src/parallel/stderr.rs

```diff
@@ -0,0 +1,47 @@
+/// Helpers functions for [ChildStderr].
+use std::process::ChildStderr;
+
+use crate::{Error, ErrorKind};
+
+#[cfg(all(not(unix), not(windows)))]
+compile_error!("Only unix and windows support non-blocking pipes! For other OSes, disable the parallel feature.");
+
+pub fn bytes_available(stderr: &mut ChildStderr) -> Result<usize, Error> {
+    let mut bytes_available = 0;
+    #[cfg(windows)]
+    {
+        use crate::windows::windows_sys::PeekNamedPipe;
+        use std::os::windows::io::AsRawHandle;
+        use std::ptr::null_mut;
+        if unsafe {
+            PeekNamedPipe(
+                stderr.as_raw_handle(),
+                null_mut(),
+                0,
+                null_mut(),
+                &mut bytes_available,
+                null_mut(),
+            )
+        } == 0
+        {
+            return Err(Error::new(
+                ErrorKind::IOError,
+                format!(
+                    "PeekNamedPipe failed with {}",
+                    std::io::Error::last_os_error()
+                ),
+            ));
+        }
+    }
+    #[cfg(unix)]
+    {
+        use std::os::unix::io::AsRawFd;
+        if unsafe { libc::ioctl(stderr.as_raw_fd(), libc::FIONREAD, &mut bytes_available) } != 0 {
```

Comment:
Stackoverflow says some implementation is broken and fail none-the-less.

I'd like it to fallback to reserving 100B on failure, and have an atomic for checking is the impl is buggy/supported.

Reply:
Ugh, ok.

I'm also noticing that I'm getting spurious 0-bytes available on Linux (Ubuntu 20.04 in WSL) as well...

Reply:
I'm also not a fan of disabling this across the entire process after a single pipe fails, so I've added a per-pipe flag if it fails or returns 0.